src/core/text/TextStyle.js

```diff
@@ -0,0 +1,315 @@
+ * @param [style.lineHeight] {number} The line height, a number that represents the vertical space that a letter uses
+ * @param [style.dropShadow=false] {boolean} Set a drop shadow for the text
+ * @param [style.dropShadowColor='#000000'] {string} A fill style to be used on the dropshadow e.g 'red', '#00FF00'
+ * @param [style.dropShadowAngle=Math.PI/4] {number} Set a angle of the drop shadow
+ * @param [style.dropShadowDistance=5] {number} Set a distance of the drop shadow
+ * @param [style.dropShadowBlur=0] {number} Set a shadow blur radius
+ * @param [style.padding=0] {number} Occasionally some fonts are cropped on top or bottom. Adding some padding will
+ *      prevent this from happening by adding padding to the top and bottom of text height.
+ * @param [style.textBaseline='alphabetic'] {string} The baseline of the text that is rendered.
+ * @param [style.lineJoin='miter'] {string} The lineJoin property sets the type of corner created, it can resolve
+ *      spiked text issues. Default is 'miter' (creates a sharp corner).
+ * @param [style.miterLimit=10] {number} The miter limit to use when using the 'miter' lineJoin mode. This can reduce
+ *      or increase the spikiness of rendered text.
+ */
+function TextStyle(style)
+{
+    EventEmitter.call(this);
+    var properties = Object.assign({}, this._defaults, style);
+    for (var property in properties)
+    {
+        this[property] = properties[property];
+    }
+}
+
+TextStyle.prototype = Object.create(EventEmitter.prototype);
+TextStyle.prototype.constructor = TextStyle;
+module.exports = TextStyle;
+
+// Default settings. Explained in the constructor.
+TextStyle.prototype._defaults = {
+    align: 'left',
+    dropShadow: false,
+    dropShadowAngle: Math.PI / 6,
+    dropShadowBlur: 0,
+    dropShadowColor: '#000000',
+    dropShadowDistance: 5,
+    lineHeight: null,
+    lineJoin: 'miter',
+    fill: 'black',
+    font: 'bold 20pt Arial',
+    miterLimit: 10,
+    padding: 0,
+    stroke: 'black',
+    strokeThickness: 0,
+    textBaseline: 'alphabetic',
+    wordWrap: false,
+    wordWrapWidth: 100
+};
+
+/**
+ * Creates a new TextStyle object with the same values as this one.
+ * Note that the only the properties of the object are cloned, not its event emitter.
+ *
+ * @return {PIXI.TextStyle}
+ */
+TextStyle.prototype.clone = function ()
+{
+    var clonedProperties = {};
+    for (var key in this._defaults)
+    {
+        clonedProperties[key] = this[key];
+    }
+    return new TextStyle(clonedProperties);
+};
+
+/**
+ * Create setters and getters for each of the style properties. Converts colors where necessary.
+ * Any set operation will emit a styleChanged event.
+ */
+Object.defineProperties(TextStyle.prototype, {
+    font: {
+        get: function ()
+        {
+            return this._font;
+        }, set: function (font)
+        {
+            if (this._font !== font)
+            {
+                this._font = font;
+                this.emit('styleChanged');
```

Comment:
If we're going to go that route, I think it'd be better to put it in `CONST`, since it's used in two files.